### ISSUE TITLE
Can't read from `.env` files using `AddEnvFile(string)`

### ISSUE THREAD

**Issue by YuKitsune** [YuKitsune/Configuration.Env#2]
When using the `.AddEnvFile(string)` method, specifying a `.env` file, a `FileNotFoundException` is always returned.

Example:
```cs
.ConfigureAppConfiguration((ctx, cfg) =>
  {
      cfg.AddEnvironmentVariables(prefix: "MyApp_");
      cfg.AddEnvFile(".env");
  })
```
Stacktrace:
```
System.IO.FileNotFoundException
The configuration file '.env' was not found and is not optional.
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.HandleException(ExceptionDispatchInfo info)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load(Boolean reload)
   at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load()
   at Microsoft.Extensions.Configuration.ConfigurationRoot..ctor(IList`1 providers)
   at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build()
```

**Comment by YuKitsune** [comment]
This is happening because by default, the `FileConfigurationSource` excludes dot-prefixed files, hidden files, and system files from being read into the configuration.

I've opened an issue on the .NET runtime repo asking why this is the case.
https://github.com/dotnet/runtime/issues/54712

**Comment by YuKitsune** [comment]
Fixed in #3 
The misleading error is being addressed in a PR on the .NET Core runtime PR: https://github.com/dotnet/runtime/pull/54717
